workers/unity/Packages/com.improbable.gdk.core/Systems/CleanReactiveComponentsSystem.cs

```diff
@@ -65,29 +57,31 @@ private void GenerateComponentGroups()
                     }
 
                     typesToRemove.Add(type);
-                    addRemoveComponentActionMethod.MakeGenericMethod(type).Invoke(this, null);
+                    AddRemoveComponentAction(type);
                 }
             }
         }
 
-        private void AddRemoveComponentAction<T>()
+        private void AddRemoveComponentAction(ComponentType type)
         {
-            var componentGroup = GetComponentGroup(ComponentType.ReadOnly<T>());
+            var componentGroup = GetComponentGroup(type);
             removeComponentActions.Add(() =>
             {
                 if (componentGroup.IsEmptyIgnoreFilter)
                 {
                     return;
                 }
 
+                RemoveAtEndOfTick typeInstance = (RemoveAtEndOfTick)Activator.CreateInstance(type.GetManagedType());
```

Comment:
Would be nice if we don't have to instantiate a component just to call RemoveComponent, but as a c# noob, I am not sure how we can force declaring static method when inheriting from an interface

Reply:
This line should go outside of the lambda function. We don't want to be using activator in a tight loop.

Reply:
However I think this will also cause a boxing allocation, which is further reason against this in total.

Reply:
You can't have a static method on an interface 😢 Its something I've lamented about in the past.